x-pack/plugins/ingest_manager/server/routes/agent_config/handlers.ts

```diff
@@ -91,15 +101,34 @@ export const getOneAgentConfigHandler: RequestHandler<TypeOf<
 
 export const createAgentConfigHandler: RequestHandler<
   undefined,
-  undefined,
+  TypeOf<typeof CreateAgentConfigRequestSchema.query>,
   TypeOf<typeof CreateAgentConfigRequestSchema.body>
 > = async (context, request, response) => {
   const soClient = context.core.savedObjects.client;
   const user = await appContextService.getSecurity()?.authc.getCurrentUser(request);
+  const withSysMonitoring = request.query.sys_monitoring ?? false;
   try {
-    const agentConfig = await agentConfigService.create(soClient, request.body, {
-      user: user || undefined,
-    });
+    const [agentConfig, newSysDatasource] = await Promise.all<
+      AgentConfig,
+      NewDatasource | undefined
+    >([
+      agentConfigService.create(soClient, request.body, {
+        user: user || undefined,
+      }),
+      // If needed, retrieve System package information and build a new Datasource for the system package
+      withSysMonitoring ? buildSystemDatasource(soClient).catch(() => undefined) : undefined,
```

Comment:
I'm ignoring failures in `buildSystemDatasource()` here - my thinking is that if the creation of the building of the NewDatasource object fails the overall Agent Create action should still return Success (unless there is a way to send back a "warning").
Let me know your thoughts. If the feedback is to allow it to fail, what would be the error handling for it? (backout/delete the created AgentConfig?)